Attempt to fix ci-gets-stuck in shutdown loop

Marenz opened the pull request:
In an attempt to find the dependency that causes the CI to get stuck (the _max test works so it must be a min dependency)

llucax commented on src/frequenz/sdk/microgrid/_power_distributing/_component_status/_battery_status_tracker.py:
```diff
@@ -262,7 +262,9 @@ async def _run(
         set_power_result = set_power_result_receiver
 
-        while True:
+        # To prevent entering a shutdown-loop, we explicitly check if there is
+        # an eventloop running
+        while asyncio.get_event_loop().is_running():
```
Shouldn't this stop when there is an exception (like there is no loop)? Or is this being run in some other `run_forever()` or something like that?

Marenz replied:
It's more about preventing it to start in the first place. The proper way is probably to do this check in select in the channels repo

llucax replied:
But I think it is fine, if there is no loop an exception is raised in `select()`, the issue is we are blindly looping on broken code forever.

llucax committed:
Fix __aexit__ signature in mock_microgrid

Signed-off-by: Leandro Lucarella <luca-frequenz@llucax.com>

llucax committed:
Ensure `ComponentPoolStatusTracker` is stopped properly

If the test raised an exception in the middle of the test, the
`ComponentPoolStatusTracker` was not stopped, which caused the
`ComponentPoolStatusTracker` to keep running and entering a restart loop
when the asyncio event loop was closed.

Signed-off-by: Leandro Lucarella <luca-frequenz@llucax.com>

llucax committed:
Make `run_forever()` more resilient when there is no event loop

We detect if the asyncio event loop is not running and log an error
message and exit instead of retrying.

Signed-off-by: Leandro Lucarella <luca-frequenz@llucax.com>

Marenz committed:
Forever loops: Always sleep, even if no exception occurs

We suspect that the CI is stuck because the forever loop is not sleeping when no exception occurs.

Signed-off-by: Mathias L. Baumann <mathias.baumann@frequenz.com>

llucax committed:
Use `run_forever()` instead of a custom loop

`run_forever()` is now more resilient to issues with no running asyncio
event loop, so it is better to use that instead of our custom naïve
loop.

Signed-off-by: Leandro Lucarella <luca-frequenz@llucax.com>
